k8sutils/finalizer.go

```diff
@@ -31,6 +35,25 @@ func HandleRedisFinalizer(cr *redisv1beta1.Redis, cl client.Client) error {
 	return nil
 }
 
+// HandleRedisClusterFinalizer finalize resource if instance is marked to be deleted
+func HandleRedisClusterFinalizer(cr *redisv1beta1.RedisCluster, cl client.Client) error {
+	if cr.GetDeletionTimestamp() != nil {
+		if controllerutil.ContainsFinalizer(cr, RedisClusterFinalizer) {
+			if err := finalizeRedisClusterServices(cr); err != nil {
+				return err
+			}
+			if err := finalizeRedisClusterPVC(cr); err != nil {
+				return err
+			}
+			controllerutil.RemoveFinalizer(cr, RedisClusterFinalizer)
+			if err := cl.Update(context.TODO(), cr); err != nil {
+				return err
+			}
+		}
```

Comment:
Some additional logging of what's going on here could be handy to understand why resources haven't been deleted.

Reply:
I agree it will be better with logging.

PS: Thanks a lot for the review